src/libstd/sys/vxworks/fast_thread_local.rs

```diff
@@ -2,7 +2,27 @@
 #![unstable(feature = "thread_local_internals", issue = "0")]
 
 pub unsafe fn register_dtor(t: *mut u8, dtor: unsafe extern "C" fn(*mut u8)) {
+    use crate::mem;
     use crate::sys_common::thread_local::register_dtor_fallback;
+    extern "C" {
+        #[linkage = "extern_weak"]
+        static __dso_handle: *mut u8;
+        #[linkage = "extern_weak"]
+        static __cxa_thread_atexit_impl: *const libc::c_void;
+    }
+    if !__cxa_thread_atexit_impl.is_null() {
+        type F = unsafe extern "C" fn(
+            dtor: unsafe extern "C" fn(*mut u8),
+            arg: *mut u8,
+            dso_handle: *mut u8,
+        ) -> libc::c_int;
+        mem::transmute::<*const libc::c_void, F>(__cxa_thread_atexit_impl)(
```

Comment:
I'm a bit confused by this. 

__cxa_thread_atexit_impl is expected to be a function. I.e. its "value" is a bunch of code. The *address* of __cxa_thread_exit_impl (*not* the "value" of __cxa_thred_atexit_impl) is a pointer to a function. Don't we need to take the address of __cxa_thread_atexit_impl before transmuting that to a function pointer?

Well I looked into it and it appears that "weak" is completely messed up in Rust.

See https://github.com/rust-lang/rust/issues/29603

especially this comment

https://github.com/rust-lang/rust/issues/29603#issuecomment-357446821

I hate to rely on a broken feature. The reason we decided to use weak was to support older VxWorks releases, right? I suppose that's a good reason.

But can you please check that this actually works as expected?